src/devices/Hx711/HX711.cs

```diff
@@ -0,0 +1,259 @@
+                _referenceUnitList.Add(referenceValue);
+                ReferanceUnit = _referenceUnitList.Average();
+
+                _isCalibrated = true;
+            }
+        }
+
+        /// <summary>
+        /// If you already know the reference unit between the Hx711 value and grams, you can set it and skip the calibration.
+        /// </summary>
+        /// <param name="referenceUnit">Conversion ratio between Hx711 units and grams</param>
+        public void SetReferenceUnit(double referenceUnit)
+        {
+            lock (_readLock)
+            {
+                ReferanceUnit = referenceUnit;
+
+                // Calibration no longer required
+                _isCalibrated = true;
+            }
+        }
+
+        /// <summary>
+        /// Read the weight from the HX711 through channel A to which the load cell is connected,
+        /// range and precision depend on load cell connected.
+        /// </summary>
+        /// <param name="numberOfReads">Number of readings to take from which to average, to get a more accurate value.</param>
+        /// <returns>Return a weigh read from HX711</returns>
+        public Mass GetWeight(int numberOfReads = 3)
+        {
+            if (!_isCalibrated)
+            {
+                throw new HX711CalibrationNotDoneException();
+            }
+
+            // Lock is internal in fisical read
+            var value = GetNetWeight(numberOfReads);
+            return Mass.FromGrams(Math.Round(value / ReferanceUnit, digits: 0));
+        }
+
+        /// <summary>
+        /// Sets tare for channel A for compatibility purposes
+        /// </summary>
+        /// <param name="numberOfReads">Number of readings to take from which to average, to get a more accurate value.</param>
+        public void Tare(int numberOfReads = 15)
+        {
+            lock (_readLock)
+            {
+                if (!_isCalibrated)
+                {
+                    throw new HX711CalibrationNotDoneException();
+                }
+
+                _tareValue = GetNetWeight(numberOfReads);
+            }
+        }
+
+        /// <summary>
+        /// Power up HX711 and set it ready to work
+        /// </summary>
+        public void PowerUp()
+        {
+            // Wait for and get the Read Lock, incase another thread is already
+            // driving the HX711 serial interface.
+            lock (_readLock)
+            {
+                // Lower the HX711 Digital Serial Clock (PD_SCK) line.
+                // Docs says "When PD_SCK Input is low, chip is in normal working mode."
+                // page 5
+                // https://html.alldatasheet.com/html-pdf/1132222/AVIA/HX711/573/5/HX711.html
+                _gpioController.Write(_pinPD_Sck, PinValue.Low);
+
+                // Wait 100µs for the HX711 to power back up.
+                DelayHelper.DelayMicroseconds(microseconds: 100, allowThreadYield: true);

```

Comment:
why the delay in the lock?

Reply:
Delay is necessary because otherwise if the signal is not maintained for 100µs, maybe because a peak comes from outside, the sensor would not switch on.